jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/StepsRunner.java

```diff
@@ -423,6 +426,21 @@ private void buildImages() {
             });
   }
 
+  private void buildManifestList() {
+    ProgressEventDispatcher.Factory childProgressDispatcherFactory =
+        Verify.verifyNotNull(rootProgressDispatcher).newChildProducer();
+
+    results.manifestListOrSingleManifest =
+        executorService.submit(
+            () -> {
+              List<Future<Image>> builtImages = new ArrayList<>();
+              builtImages.addAll(results.builtImagesAndBaseImages.get().keySet());
+              return new BuildManifestListStep(
+                      buildContext, childProgressDispatcherFactory, realizeFutures(builtImages))
+                  .call();
+            });
```

Comment:
This is confusing

```suggestion
            () ->
              new BuildManifestListStep(
                      buildContext,
                      childProgressDispatcherFactory,
                      realizeFutures(results.builtImagesAndBaseImages.get().keySet()))
                  .call());
```

Reply:
We need to convert the `keySet()` to an ArrayList before using the `realizeFutures` .

`   new BuildManifestListOrSingleManifestStep(
                        buildContext,
                        childProgressDispatcherFactory,
                        realizeFutures(
                            new ArrayList<>(results.builtImagesAndBaseImages.get().keySet())))
                    .call());

`
This might be better

Reply:
Aha, I think we should change `realizeFutures()` instead.
```java
  private static <E> List<E> realizeFutures(List<Future<E>> futures)
```
```java
  private static <E> List<E> realizeFutures(Collection<Future<E>> futures)
```
`Set`, `List`, `Vector`, etc. all implement `Collection`.

Reply:
Sounds Good I will do a small PR to change `realizeFutures`

Reply:
It will be a one-line change. If you prefer, you can do it here.